### PR TITLE
fix(nonentitled-section): add a11y test, fix issue with repeated alt attribute value

### PR DESCRIPTION
**Pull request by jendowns**
## Proposed changes

- add a11y test
- remove snapshots
- fix issue with repeated alt attribute value + add `aria-hidden` (relevant text already exists in the enclosing `a` element)
- remove unused prop `catalogId`

### CODE REVIEW

File: src/components/NonEntitledSection/NonEntitledSection.js [hunk 1]
```diff
@@ -94,9 +94,6 @@ NonEntitledSection.propTypes = {
   /** @type {string} Section description. */
   description: PropTypes.string,
 
-  /** @type {string} Catalog id. */
-  catalogId: PropTypes.string,
-
```
Comment (jendowns) [inline src/components/NonEntitledSection/NonEntitledSection.js]: FYI this prop does not appear to be used anywhere? 👀

**Comment by jendowns** [comment]
Hey @SimonFinney just a reminder that this one is ready to review, with updates 👍 Also removes an (apparently?) unused prop.

**Comment by jendowns** [comment]
@SimonFinney Just a reminder that PR is ready to review. 👍

File: src/components/NonEntitledSection/__tests__/NonEntitledSection.spec.js [hunk 2]
```diff
@@ -3,28 +3,119 @@
...
+
+  test('should apply a custom class name', () => {
+    const { container } = render(
+      <NonEntitledSection
+        title="test title"
+        subTitle="test subtitle"
+        className="custom-class"
+      />
+    );
+    expect(container.firstElementChild).toHaveClass('custom-class');
```
Comment (diego-codes) [inline src/components/NonEntitledSection/__tests__/NonEntitledSection.spec.js]: I would have written this differently to rely less on the rendered structure from the testing library.

```suggestion
    expect(container.querySelector('.custom-class')).toBeInTheDocument();
```

**Pull request merged**